rules/autoconsent/espace-personnel-agirc-arrco-fr.json

```diff
@@ -0,0 +1,26 @@
+{
+    "name": "espace-personnel.agirc-arrco.fr",
+    "prehideSelectors": [
+        ".cdk-overlay-container"
+    ],
+    "detectCmp": [
+        {
+            "exists": ".cdk-overlay-container"
```

Comment:
It seems like this selector is [very common](https://publicwww.com/websites/%22cdk-overlay-container%22/), but not always associated with a cookie pop-up. Do you want to make it more specific?

Reply:
@muodov I'm little bit confused with the conceptional difference of `detectCmp` and `detectPopup`. I believed that `detectCmp` should always return `true` unlike `detectPopup`. However, I found that `app-esaa-cookie-component` is not always present. Do you think if there's a better way to describe `detectCmp`.

Reply:
I agree, the difference is subtle and tricky to explain.
`detectCmp` is supposed to return true when we're sure that this CMP is used by the site. Main purpose of `detectPopup` is to detect if the pop-up is actually open.
So for example, after you accept/reject cookies, and then reload, `detectPopup` should return false, but `detectCmp` can still return true.

The way it works, we first look for `detectCmp`, and if we find something, we _stop looking for other rules_. Therefore, we should avoid false positives in `detectCmp`, because it's possible some other CMP is still on the page, but we'll not find it.

In practice, most rules use `exists` in `detectCmp`, and `visible` in `detectPopup`, with the same selector.

It's still confusing though, so if you have ideas on how to make it better, please let us know!

Reply:
Thank you for your explanation. :+1: It helped me to understand the current stance better.